packages/ui/src/components/molecules/PopUp/NewPopUp.tsx

```diff
@@ -0,0 +1,161 @@
+/*
+ * Axelor Business Solutions
+ *
+ * Copyright (C) 2023 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or  modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, {useMemo} from 'react';
+import {StyleSheet, View, Modal} from 'react-native';
+import {useThemeColor} from '../../../theme/ThemeContext';
+import {Card, Text, Icon} from '../../atoms';
+import Button, {ButtonProps} from '../Button/Button';
+
+interface ButtonConfig extends ButtonProps {
+  hide?: boolean;
+}
+interface CancelButtonConfig extends ButtonConfig {
+  showInHeader?: boolean;
+}
+
+interface PopUpProps {
+  style?: any;
+  visible: boolean;
+  title?: string;
+  children: any;
+  cancelButtonConfig?: CancelButtonConfig;
+  confirmButtonConfig?: ButtonConfig;
+  translator?: (key: string) => string;
+}
+
+const PopUp = ({
+  style,
+  visible,
+  title,
+  children,
+  cancelButtonConfig,
+  confirmButtonConfig,
+  translator,
+}: PopUpProps) => {
+  const Colors = useThemeColor();
+
+  const _cancelButtonConfig = useMemo(() => {
+    if (!cancelButtonConfig) {
+      return null;
+    }
+
+    return {
+      ...cancelButtonConfig,
+      title: cancelButtonConfig.title ?? translator('Base_Cancel'),
+      color: cancelButtonConfig.color ?? Colors.errorColor,
+      iconName: cancelButtonConfig.iconName ?? 'times',
+      hide: cancelButtonConfig.hide ?? false,
+      showInHeader: cancelButtonConfig.showInHeader ?? false,
+      width: cancelButtonConfig.width ?? 115,
+    };
```

Comment:
Simplify here by defining default config which will be rewritten by config given in props

```
    return {
      title: translator('Base_Cancel'), // Please give default value to translator or it might throw an error here : value => value
      color: Colors.errorColor,
      iconName: 'times',
      hide: false,
      showInHeader: false,
      width: 115, // No magic number please
      ...cancelButtonConfig,
    };
```